### PR TITLE
Increase testing coverage for Upload Manager

### PR DESCRIPTION
**Pull request by peterjan**
This PR increases (unit) test coverage of the upload manager, implicitly also covering a large chunk of the download manager as well. It adds tests for packed uploads and migrating shards, currently those tests are pretty basic, not sure it adds a lot of value to try and add edge cases at this point. I'd rather assert things like refresh uploaders/downloaders works as we expect it since that's a pretty dangerous spot. Adding basic overdrive coverage is another thing I'd like to add soon.

### CODE REVIEW

File: worker/upload_test.go [hunk 1]
```diff
@@ -127,14 +133,250 @@ func TestUploadDownload(t *testing.T) {
 	}
 }
 
-func testParameters(bucket, path string) uploadParameters {
+func TestUploadPackedSlab(t *testing.T) {
+	// create upload params
```
Comment (ChrisSchinnerl) [inline worker/upload_test.go:137]: Would be nice to deduplicate the first 20 or so lines of this test. Maybe into something like a mockWorker.

File: worker/mocks_test.go [hunk 2]
```diff
@@ -42,8 +44,17 @@ type (
 	mockMemoryManager struct{}
 
 	mockObjectStore struct {
-		mu      sync.Mutex
-		objects map[string]map[string]object.Object
+		mu           sync.Mutex
+		objects      map[string]map[string]object.Object
+		partials     map[string]mockPackedSlab
+		bufferIDCntr uint // allows marking packed slabs as uploaded
```
Comment (ChrisSchinnerl) [inline worker/mocks_test.go:50]: I think this is only set but never really used?

Comment (peterjan) [reply worker/mocks_test.go:50]: It is used actually. I use it as the bufferID which I need to make `MarkPackedSlabsUploaded` work.

File: worker/upload_test.go (hunk 1 above)
Comment (peterjan) [reply worker/upload_test.go:137]: That was coming in the next one but I can add it here.

File: worker/mocks_test.go [hunk 3]
```diff
@@ -113,33 +163,111 @@ func (os *mockObjectStore) Object(ctx context.Context, bucket, path string, opts
...
 func (os *mockObjectStore) Slab(ctx context.Context, key object.EncryptionKey) (object.Slab, error) {
-	return object.Slab{}, nil
+	os.mu.Lock()
+	defer os.mu.Unlock()
+
+	for _, objects := range os.objects {
+		for _, object := range objects {
+			for _, slab := range object.Slabs {
+				if slab.Slab.Key.String() == key.String() {
+					return slab.Slab, nil
```
[the remark this answers is not in the document]
Comment (peterjan) [reply worker/mocks_test.go]: It wasn't too bad imo but I added `os.forEachObject` which I could re-use 3 times.

**Review by ChrisSchinnerl — APPROVED** [review]
